MSW mock: `useExamples` ignores property-level `example` after OAS 3.0 → 3.1 upgrade (regression)

### Description

When `output.mock.useExamples: true` is enabled and the input is an **OpenAPI 3.0** spec with `example` (singular) on a property, the example is **silently dropped** from the generated MSW mock and the value falls back to faker.

This is a regression: `useExamples` was introduced in #963 ("feat(msw): support using OpenAPI example/examples fields as MSW values") with explicit support for property-level `example`, and worked for OAS 3.0 inputs at the time. It was broken when `@scalar/openapi-parser` was wired into orval (#2614, OpenAPI 3.1 upgrade), because that integration unconditionally upgrades 3.0 → 3.1 and rewrites property-level `example` into an `examples` array — and the scalar mock getter never learned about the new shape.

Reproduced end-to-end against the latest published `orval@8.9.0` (npm).

### Output client

other

(MSW mock — issue is in `@orval/mock` and is independent of the API client.)

### Configuration (orval.config)

```ts
import { defineConfig } from 'orval';

export default defineConfig({
  repro: {
    input: { target: './spec.yaml' },
    output: {
      target: './generated/api.ts',
      mode: 'tags-split',
      mock: {
        type: 'msw',
        useExamples: true,
      },
    },
  },
});
```

### Environment

```
System:
  OS: macOS 15.7.2
  CPU: arm64 Apple M3 Pro
  Shell: zsh
npmPackages:
  orval: 8.9.0   # latest at time of filing
  msw: ^2
  @faker-js/faker: latest
```

Also confirmed at the unit-test level on `master` (current `master` is at 8.9.0) — see "Additional context".

### Expected behavior

`getGetThingsResponseMock()` returns `slug: "relaxation"`.

### Actual behavior

`slug` falls back to faker — the configured example value is silently dropped.

Generated `default.msw.ts` (orval 8.9.0):

```ts
/**
 * Generated by orval v8.9.0 🍺
 */
export const getGetThingsResponseMock = (
  overrideResponse: Partial<Extract<GetThings200, object>> = {},
): GetThings200 => ({
  data: Array.from(
    { length: faker.number.int({ min: 1, max: 10 }) },
    (_, i) => i + 1,
  ).map(() => ({
    slug: faker.string.alpha({ length: { min: 10, max: 20 } }), // ← should be "relaxation"
  })),
  ...overrideResponse,
});
```

### OpenAPI document (minimal, if applicable)

```yaml
openapi: '3.0.0'
info:
  title: Reproduction
  version: '1.0'
paths:
  /things:
    get:
      operationId: getThings
      responses:
        '200':
          description: ok
          content:
            application/json:
              schema:
                type: object
                required: [data]
                properties:
                  data:
                    type: array
                    items:
                      type: object
                      required: [slug]
                      properties:
                        slug:
                          type: string
                          example: 'relaxation'
```

### Additional context

#### Root cause

orval calls `upgrade()` from `@scalar/openapi-parser` for every spec (`packages/orval/src/import-specs.ts:63`). That helper delegates to `@scalar/openapi-upgrader`, which rewrites property-level `example` into an `examples` array and **deletes** the original singular field for OAS 3.0 inputs:

```js
// @scalar/openapi-upgrader@0.2.2
// dist/3.0-to-3.1/upgrade-from-three-to-three-one.js
if (schema.example !== undefined && !isInsideExamplesMap) {
    if (isSchemaPath(path)) {
        schema.examples = [schema.example];
    }
    ...
    delete schema.example;
}
```

So by the time the spec reaches the mock generator, the property no longer has `example` — it has `examples: [<value>]`.

#### Where the bug lives

The scalar (property-level) handler in `packages/mock/src/faker/getters/scalar.ts` only checks `item.example`, not `item.examples`:

```ts
// packages/mock/src/faker/getters/scalar.ts:101-112
if (
  (context.output.override.mock?.useExamples ||
    safeMockOptions.useExamples) &&
  item.example !== undefined
) {
  return {
    value: JSON.stringify(item.example),
    imports: [],
    name: item.name,
    overrided: true,
  };
}
```

For comparison, the response-level handler in `packages/mock/src/msw/mocks.ts` already reads both forms via the `getExampleEntries` helper:

```ts
// packages/mock/src/msw/mocks.ts:210-216
if (context.output.override.mock?.useExamples || mockOptions?.useExamples) {
  const exampleValue = unwrapExampleValue(
    example ??
      originalSchema?.example ??
      getExampleEntries(examples)[0] ??
      getExampleEntries(originalSchema?.examples)[0],
  );
  ...
}
```

This asymmetry between scalar and response handlers is the bug.

#### Unit-level reproduction (against `master` @ 8.9.0)

Adding the following to `packages/mock/src/faker/getters/scalar.test.ts` fails on the unmodified scalar getter:

```ts
describe('getMockScalar (property-level examples array — post-upgrader shape)', () => {
  const baseArg = {
    imports: [],
    operationId: 'test-operation',
    tags: [],
    existingReferencedProperties: [],
    splitMockImplementations: [],
    mockOptions: { useExamples: true },
    context: { output: { override: {} } } as ContextSpec,
  };

  it('should use examples[0] for a string property when useExamples is true', () => {
    const result = getMockScalar({
      ...baseArg,
      item: {
        type: 'string' as const,
        name: 'slug',
        examples: ['relaxation'],
      } as Parameters<typeof getMockScalar>[0]['item'],
    });

    expect(result.value).toBe('"relaxation"');
  });
});
```

```
AssertionError: expected 'faker.string.alpha()' to be '"relaxation"'
- Expected: ""relaxation""
- Received: "faker.string.alpha()"
```

#### Suggested fix

Treat `item.examples[0]` as a fallback when `item.example` is absent in the scalar getter, mirroring the response-level handler:

```ts
const useExamplesEnabled =
  context.output.override.mock?.useExamples || safeMockOptions.useExamples;
const propExample =
  item.example !== undefined
    ? item.example
    : Array.isArray(item.examples) && item.examples.length > 0
      ? item.examples[0]
      : undefined;
if (useExamplesEnabled && propExample !== undefined) {
  return {
    value: JSON.stringify(propExample),
    imports: [],
    name: item.name,
    overrided: true,
  };
}
```

#### Prior art

This is the same shape of bug (post-upgrader OAS keyword change not reflected in the scalar mock getter) that was previously fixed for:

- `exclusiveMinimum` / `exclusiveMaximum` (boolean → numeric) — #3142, #3221
- `format: binary` → `contentMediaType: application/octet-stream` — #3248

#### Out of scope for this issue

OAS 3.1 specs that use `examples: [...]` (array) at the property level are also currently not picked up by the scalar getter. That is a pre-existing scope gap (the scalar getter never learned about the `examples` array form), not a regression — happy to file a separate enhancement issue if maintainers prefer.

Happy to send a PR for this OAS 3.0 regression.

--- a/packages/mock/src/faker/getters/scalar.test.ts
+++ b/packages/mock/src/faker/getters/scalar.test.ts
@@ -632,3 +632,32 @@ describe('getMockScalar (pattern-backed string escaping)', () => {
     );
   });
 });
+
+describe('getMockScalar (post-upgrader OAS 3.0 example handling)', () => {
+  // OAS 3.0 inputs go through @scalar/openapi-parser's upgrade(), which
+  // rewrites property-level `example: <value>` into `examples: [<value>]`
+  // and deletes the singular field. The scalar getter must read the array
+  // form so that useExamples keeps working for OAS 3.0 specs.
+  const baseArg = {
+    imports: [],
+    operationId: 'test-operation',
+    tags: [],
+    existingReferencedProperties: [],
+    splitMockImplementations: [],
+    mockOptions: { useExamples: true },
+    context: { output: { override: {} } } as ContextSpec,
+  };
+
+  it('uses examples[0] for a string property when useExamples is true', () => {
+    const result = getMockScalar({
+      ...baseArg,
+      item: {
+        type: 'string' as const,
+        name: 'slug',
+        examples: ['relaxation'],
+      } as Parameters<typeof getMockScalar>[0]['item'],
+    });
+
+    expect(result.value).toBe('"relaxation"');
+  });
+});

--- a/packages/mock/src/faker/getters/scalar.ts
+++ b/packages/mock/src/faker/getters/scalar.ts
@@ -99,16 +99,27 @@ export function getMockScalar({
   }
 
   if (
-    (context.output.override.mock?.useExamples ||
-      safeMockOptions.useExamples) &&
-    item.example !== undefined
+    context.output.override.mock?.useExamples ||
+    safeMockOptions.useExamples
   ) {
-    return {
-      value: JSON.stringify(item.example),
-      imports: [],
-      name: item.name,
-      overrided: true,
-    };
+    // OAS 3.0 inputs go through @scalar/openapi-parser's upgrade(), which
+    // rewrites property-level `example: <value>` into `examples: [<value>]`
+    // and deletes the singular field. Fall back to the array form so this
+    // option keeps working post-upgrade.
+    const propertyExample =
+      item.example === undefined
+        ? Array.isArray(item.examples) && item.examples.length > 0
+          ? item.examples[0]
+          : undefined
+        : item.example;
+    if (propertyExample !== undefined) {
+      return {
+        value: JSON.stringify(propertyExample),
+        imports: [],
+        name: item.name,
+        overrided: true,
+      };
+    }
   }
 
   const formatOverrides = safeMockOptions.format ?? {};

